app/main.py

```diff
@@ -1,25 +1,65 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: str, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: str, owner: str) -> str:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: str, value: int | str) -> Any:
+        if not isinstance(value, int):
+            raise TypeError("Value should be an integer.")
+
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError(f"Value should be between"
+                             f" {self.min_amount} and {self.max_amount}.")
+        setattr(instance, self.protected_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    weight = IntegerRange(20, 50)
+    height = IntegerRange(80, 120)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    weight = IntegerRange(50, 120)
+    height = IntegerRange(120, 220)
 
 
 class Slide:
-    pass
+    def __init__(self, name: str, limitation_class: Any) -> None:
```

Comment:
Consider using a more specific type for `limitation_class` instead of `Any`. You could use `Type[SlideLimitationValidator]` to indicate that it should be a class type derived from `SlideLimitationValidator`.